feat: add `symbol/match` (#8547)
stdlib-js/stdlib: lib/node_modules/@stdlib/symbol/match/README.md

<!--

@license Apache-2.0

Copyright (c) 2025 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# MatchSymbol

> > [Symbol][mdn-symbol] which specifies the matching of a regular expression against a string.

Reviewer: ```suggestion
> [Symbol][mdn-symbol] which specifies whether an object should be treated as a regular expression.
```

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- Package usage documentation. -->

<section class="usage">

## Usage

```javascript
var MatchSymbol = require( '@stdlib/symbol/match' );
```

#### MatchSymbol

> [`Symbol`][mdn-symbol] which specifies the matching of a regular expression against a string. Used by `String.prototype.match()`.

Reviewer: ```suggestion
[`Symbol`][mdn-symbol] which specifies whether an object should be treated as a regular expression.
```

```javascript
var s = typeof MatchSymbol;
// e.g., returns 'symbol'
```

</section>

<!-- /.usage -->

<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

## Notes

-   The [symbol][mdn-symbol] is only supported in environments which support [symbols][mdn-symbol]. In non-supporting environments, the value is `null`.

> 

Reviewer: Describe how `MatchSymbol` is used here. 1) As a method used by `String.prototype.match` to match an input string against the current object. This method is provided a single argument: a string. 2) To determine whether an object should be treated as a regular expression.

This is discussed in more depth on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/match

</section>

<!-- /.notes -->

<!-- Package usage examples. -->

<section class="examples">

## Examples

<!-- eslint no-undef: "error" -->

```javascript
var defineProperty = require( '@stdlib/utils/define-property' );
var MatchSymbol = require( '@stdlib/symbol/match' );

var obj = {};
var re = /foo/;

// Define a custom match behavior:
function customMatch() {
    return [ 'custom', 'match', 'result' ];
}

defineProperty( obj, MatchSymbol, {
    'configurable': false,
    'enumerable': false,
    'writable': false,
    'value': customMatch
});

var v = 'foobar'.match( obj );
console.log( v );
// => [ 'custom', 'match', 'result' ]

// Default match behavior:
v = 'football'.match( re );

> console.log( JSON.stringify( v ) );

Reviewer: You don't need `JSON.stringify`.
